gemfirexd/tools/src/test/java/com/pivotal/gemfirexd/internal/engine/distributed/metadata/UpdateQueryInfoInternalsTest.java

```diff
@@ -599,7 +602,580 @@ public void queryInfoObjectFromOptmizedParsedTree(QueryInfo qInfo,
     }
 
   }
-  
+
+  /**
+   * Tests if the where clause containing multiple fields with primary keys and extra
+   * conditions
+   */
+  public void testUpdatePutConvertibleExtraAnds() throws Exception {
+    Connection conn = getConnection();
+    Statement s = conn.createStatement();
+    s.execute("create table src_tab (col1 varchar(10), col2 int , " +
+        "col3 int, col4 int, PRIMARY KEY (col1, col2)) partition by column (col1)");
+    String query = "update src_tab set col3=? where col2=? and col1=? and col3=?";
+    GemFireXDQueryObserver old = null;
```

Comment:
lets try to update some other column than in the predicate. pick multiple column update along with 'where timestamp < CURRENT_TIME' kind of additional where clause.